src/scenes/Event/scenes/oracle.js

```diff
@@ -589,74 +539,77 @@ export default class OraclePage extends React.Component {
     if (token === Token.Qtum && status === OracleStatus.WaitResult && resultSetterQAddress !== lastUsedAddress) {
       return {
         disabled: true,
-        message: <FormattedMessage
-          id="oracle.cOracleDisabledText"
-          defaultMessage="You are not the result setter for this Event. You must wait until they set the result, or until the Open Result Set start time begins."
-        />,
+        id: 'oracle.cOracleDisabledText',
+        message: 'You are not the result setter for this Event. You must wait until they set the result, or until the Open Result Set start time begins.',
         warningTypeClass: EventWarningType.Info,
       };
     }
 
-    // Did not select a result
-    if (!(token === Token.Bot && status === OracleStatus.WaitResult) && currentOptionIdx === -1) {
-      return {
-        disabled: true,
-        message: <FormattedMessage
-          id="oracle.selectResultDisabledText"
-          defaultMessage="Please click and select one of the options."
-        />,
-        warningTypeClass: EventWarningType.Info,
-      };
-    }
-
-    // Did not enter an amount
-    if (status === OracleStatus.Voting && (voteAmount <= 0 || Number.isNaN(voteAmount))) {
+    // Trying to set result or vote when not enough QTUM or BOT
+    const totalQtum = _.sumBy(walletAddresses, (wallet) => wallet.qtum ? wallet.qtum : 0);
+    const filteredAddress = _.filter(walletAddresses, { address: lastUsedAddress });
+    const currentBot = filteredAddress.length > 0 ? filteredAddress[0].bot : 0; // # of BOT at currently selected address
+    if ((token === Token.Bot || (token === Token.Qtum && status === OracleStatus.OpenResultSet))
+      && (totalQtum < maxTransactionFee && currentBot < oracle.consensusThreshold)
```

Comment:
`currentBot < oracle.consensusThreshold` is only applicable when the user is setting the result. `token === Token.Bot` + `status === OracleStatus.Voting` (which is Voting status) should check `currentBot < voteAmount`.

Rules for allowing action by status:
- Bet: `totalQtum >= voteAmount + maxTransactionFee`
- SetResult: `totalQtum >= maxTransactionFee` + `currentBot >= consensusThreshold`
- Vote: `totalQtum >= maxTransactionFee` + `currentBot >= voteAmount`
- Finalize: `totalQtum >= maxTransactionFee`
- Withdraw: `totalQtum >= maxTransactionFee`